core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -503,6 +550,84 @@ public boolean namespaceExists(Namespace namespace) {
     return JdbcUtil.namespaceExists(catalogName, connections, namespace);
   }
 
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
+    int deletedRecords =
+        execute(
+            JdbcUtil.DROP_SQL,
+            catalogName,
+            JdbcUtil.namespaceToString(identifier.namespace()),
+            identifier.name(),
+            "VIEW");
+
+    if (deletedRecords == 0) {
+      LOG.info("Skipping drop, view does not exist: {}", identifier);
+      return false;
+    }
+
+    LOG.info("Dropped view: {}", identifier);
+    return true;
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    if (!namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    return fetch(
+        row ->
+            JdbcUtil.stringToTableIdentifier(
+                row.getString(JdbcUtil.TABLE_NAMESPACE), row.getString(JdbcUtil.TABLE_NAME)),
+        JdbcUtil.LIST_SQL,
+        catalogName,
+        JdbcUtil.namespaceToString(namespace),
+        "VIEW");
+  }
+
+  @Override
+  public void renameView(TableIdentifier from, TableIdentifier to) {
+    if (!viewExists(from)) {
+      throw new NoSuchViewException("View does not exist");
+    }
+
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", to.namespace());
+    }
+
+    if (tableExists(to)) {
```

Comment:
I feel like this check is incomplete. Shouldn't we be checkin for table or view existence?  (Same is probably true for rename table).

Reply:
I did that initially, but the tests (provided by the `TestCatalog` abstract class) fail as they don't expect `NoSuchTableException` or `NoSuchViewException`. Let me try again.

It's what I said during the last community meeting: the tests should covered this use case (testing both view and table exists before creating a table or view) and all catalogs should be updated to check that. If so, I propose to create another PR (and corresponding issue) about that.

Reply:
`InMemoryCatalog` checks already for table/view existence before renaming: https://github.com/apache/iceberg/blob/2446cee5cf0ad93a2be9a68f0b2f7f6fa6edb865/core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java#L347-L367